tortoise/rerun_test.go

```diff
@@ -50,3 +50,99 @@ func TestRecoverState(t *testing.T) {
 	require.Equal(t, cfg.MeshVerified, old)
 	require.Equal(t, last.Sub(1), verified)
 }
+
+func TestRerunDistanceVoteCounting(t *testing.T) {
+	genesis := types.GetEffectiveGenesis()
+	ctx := context.Background()
+	const size = 10
+	s := sim.New(sim.WithLayerSize(size))
+	s.Setup()
+
+	cfg := defaultTestConfig()
+	cfg.LayerSize = size
+	cfg.Hdist = 1
+	cfg.Zdist = 1
+
+	tortoise := tortoiseFromSimState(s.GetState(0), WithLogger(logtest.New(t)), WithConfig(cfg))
+	var last, verified types.LayerID
+
+	const firstBatch = 5
+	misverified := genesis.Add(firstBatch)
+	for _, last = range sim.GenLayers(s,
+		sim.WithSequence(firstBatch, sim.WithEmptyInputVector()),
+		// in this layer voting is malicious, and it will make that misverified layer will be all invalid
+		sim.WithSequence(1, sim.WithVoteGenerator(gapVote)),
+		// in this layer we skip previously malicious voting
+		sim.WithSequence(1, sim.WithVoteGenerator(gapVote)),
+		sim.WithSequence(5),
+	) {
+		_, verified, _ = tortoise.HandleIncomingLayer(ctx, last)
+	}
+	require.Equal(t, last.Sub(1), verified)
+
+	blocks, err := s.GetState(0).MeshDB.LayerBlockIds(misverified)
+	require.NoError(t, err)
+	for _, block := range blocks {
+		validity, err := s.GetState(0).MeshDB.ContextualValidity(block)
+		require.NoError(t, err)
+		require.False(t, validity, "validity for block %s", block)
+	}
+	require.NoError(t, tortoise.rerun(ctx))
```

Comment:
can you test here that rerun did change our opinions on the mis-verified layer? 

as the code is now, i am not sure it's the call tortose.rerun() or the extra layer votes later that made the difference.

Reply:
sure, will reorder checks